src/querystring.rs

```diff
@@ -0,0 +1,79 @@
+use crate::{error::Error, Context};
+use http::StatusCode;
+use serde::Deserialize;
+
+/// An extension trait for `Context`, providing query string deserialization.
+pub trait ExtractQuery<'de> {
+    fn query<T: Deserialize<'de>>(&'de self) -> Result<T, Error>;
+}
+
+impl<'de, Data> ExtractQuery<'de> for Context<Data> {
+    fn query<T: Deserialize<'de>>(&'de self) -> Result<T, Error> {
```

Comment:
Same as above.

Reply:
See response above. `serde::Deserialize` is preferred over `serde::de::Deserialize` in Rust 2018 imports.